### ISSUE TITLE
Type check global after function definition

### ISSUE THREAD

**Issue by spkersten** [python/mypy#538]
``` python
import typing

x = 'Hagar Qim'

def f() -> None:
  global x, y
  x = 1  # Type error reported
  y = 2  # Ignored

y = 'Mnajdra'
```

No type error is reported for the assignments to `y`.

**Comment by ethanhs** [comment]
The original example errors on assignment to y correctly (`Incompatible types in assignment (expression has type "int", variable has type "str")`). I think this can be closed.

**Comment by ilevkivskyi** [comment]
Yes, but it infers the type of ``y`` from the second assignment, not as Jukka proposed:
> It should infer the type of y from the first assignment (int in the example).

I leave this up to him to close.

**Comment by ethanhs** [comment]
Hm, I missed that, but I don't think that is _always_ correct. If f is never executed then the type should be a string for all intents and purposes. The correctness really depends on whether f is run before the global assignment (str type). Handling something like that might be very complicated however.

**Comment by JukkaL** [comment]
The order in which code is executed at runtime is not important for mypy when inferring types. In this case, the consistent behavior would be to infer the type using assignment that comes first textually. However, this is not a high-priority thing to fix.

**Comment by davfsa** [comment]
This issue might have gotten fixed down the line, as it now errors (tested on mypy 1.0.0):
```
# mypy a.py
foo.py:7: error: Incompatible types in assignment (expression has type "int", variable has type "str")  [assignment]
foo.py:8: error: Incompatible types in assignment (expression has type "int", variable has type "str")  [assignment]
Found 2 errors in 1 file (checked 1 source file)
```

Is this considered as being fixed?

**Comment by hauntsaninja** [comment]
https://github.com/python/mypy/issues/538#issuecomment-312476241 has the current status of the issue. Given that mypy does report a type error, I don't think it's super important to fix / I haven't seen other issue reports about this. But still a thing that could be improved.
